### ISSUE TITLE
Stopping Criteria

### ISSUE THREAD

**Issue by AminRoma** [dirkschumacher/ompr#290]
Is there any way to set the optimization problem stops after a certain amount of time or reaching a certain optimality gap? Is there any way to set these parameters before running our code?

**Comment by prof-anderson** [comment]
Yes but options vary by solver used.  For example, using the symphony solver, this will call a gap limit of 1.5%.  I've commented out terminating after the first feasible solution.
`res_Mail1 <- solve_model(Mail1, 
                        with_ROI(solver = "symphony",
                   verbosity=-1, gap_limit=1.5))
#                   verbosity=-1, first_feasible=TRUE))
`
Note that ompr doesn't seem to be able to interpret the various stopping conditions in terms of a correct status and instead returns a status of infeasible. (This is a known issue discussed elsewhere.)

Checking the verbose output will confirm that it is feasible though.

Note that other solvers have other options.  
  
Tim

**Comment by AminRoma** [comment]
Thanks for your response. I am going to test it on my code. Do we have the same syntax for glpk solver as well? I searched for it but I could not find a similar example.

**Comment by prof-anderson** [comment]
Each solver has their own choices for optional parameters as demonstrated by glpk and symphony using verbose and verbosity respectively.  

From a quick search, a stack overflow response from Florian says that the ROI.plugin.glpk actually uses the rglpk package internally and therefore can only relay on the glpk control parameters that rglpk can accept.  The result is that roi.plug.glpk accepts a limited number of control options.  I think tm_limit should work for a time limit in milliseconds but I haven't tested it.
[Stack Overflow post about roi.plugin.glpk](https://stackoverflow.com/questions/57407103/r-how-to-hand-over-mip-gap-option-to-glpk-in-roi) 

On the MIPs that I've tested through ompr and their respective roi plugins, I haven't seen a big difference between glpk, LPSolve, and Symphony performance.  Symphony seems to provide more control options than glpk.

I was hoping to test gurobi but can't get the third party roi plugin working for gurobi.

**Comment by AminRoma** [comment]
Thanks for your answer. For future use, I only add that the unit of time we put as an argument in tm_limit is millisecond.

**Comment by karthikkommalapati** [comment]
Hi @prof-anderson & @AminRoma ,

Grateful if you have could please point me towards control parameter for precision, I am after 4 decimal places of precisions and not sure if **glpk** solver offers that control parameter. Tried to find the rescorce but no luck so far and this precision currently fails when I use the objective value from model one as constraint to the next model. Grateful if you could point me where I can apply this control parameter.

Kind Regards,
Karthik
